### DISCUSSION TITLE
Namespace not found error

### DISCUSSION THREAD

**Discussion by wjke-nics** [basilisp-lang/basilisp#1197] [Q&A] [answered]
Sorry to ask what might be a silly question, but I'm encountering a namespace resolution issue in my Basilisp project. My project structure is:

src/prob-l/
  ├── core.lpy
  └── model.lpy

In core.lpy:
``` clojure 
(ns prob-l.core)
```
In model.lpy:
``` clojure 
(ns prob-l.model
    :require [prob-l.core])
```

The error shows "namespace not found" when trying to require prob-l.core in model.lpy. Am I missing something in the project setup or namespace declaration?

I'm developing a probabilistic programming language based on JAX using Basilisp. Looking forward to sharing it with the community when ready!

**Comment by ikappaki** [comment, marked as answer]
Hi @wjke-nics,

this issue likely arises from using a hyphen in `src/prob-l/` instead of `src/prob_l`. Please refer to the the [Runtime/Namespace](https://basilisp.readthedocs.io/en/latest/runtime.html#namespaces) section in the documentation:

> Namespace names should match the document’s place in the source tree, with forward slash (/) characters being replaced by periods (.) and underscores (_) replaces with hyphens (-).


Additionally, since your code is in `src/`, which isn't included in Python `sys.path` by default, you'd need to include it in the `ns` declaration, e.g., `(ns src.prob-l.model ...)`, even though this is not probably what you want to do.  There is a [feature request](https://github.com/basilisp-lang/basilisp/issues/900) to specify project source locations, but it's part of broader CLI tools task that is still under planning.

Managing additional source locations is thus delegated to the Python build too being used to develop the application. For example, with the [Poetry](https://python-poetry.org/docs/) tool, you can use the the `--src` option when creating a `new` project to automatically include the `src` directory in `sys.path` when activating the virtual environment.

```ps1
# create new project with a `src` directory, which will be added to the `sys.path` after activating the environment
> poetry new prob-l --src

# add Basilisp as a dependency
> poetry add basilisp

# install project so that is available within the virtual environment
prob-l> poetry install

# activate the virtual environment
> poetry shell

(prob-l-py3.11) PS prob-l> basilisp repl
basilisp.user=>
```

I hope this helps!
